test/filtering/filter_test.py

```diff
@@ -1,30 +1,98 @@
 from otri.filtering.filter import Filter, Stream
+from unittest.mock import MagicMock
 import unittest
 
-s_1 = Stream([1, 2, 3])
-s_2 = Stream([3, 4, 5])
 
 class FilterTest(unittest.TestCase):
 
     def setUp(self):
+        self.s_A = Stream([1, 2, 3])
+        self.s_B = Stream([3, 4, 5])
+        self.s_D = Stream()
+        self.s_E = Stream()
+        self.s_F = Stream()
+        self.state = dict()
         self.f = Filter(
-            inputs=["A","B"],
-            outputs=["D","E","F"],
+            inputs=["A", "B"],
+            outputs=["D", "E", "F"],
             input_count=2,
             output_count=3
         )
+        self.f.setup([self.s_A, self.s_B], [self.s_D, self.s_E, self.s_F], self.state)
 
     def test_filter_input_number_correct(self):
-        self.assertEqual(2, len(self.f.get_inputs()))
-    
-    def test_filter_input_streams_equals(self):
-        self.assertEqual(["A","B"], self.f.get_inputs())
+        self.assertEqual(2, len(self.f.get_input_names()))
+
+    def test_filter_input_stream_names_equals(self):
+        self.assertEqual(["A", "B"], self.f.get_input_names())
 
     def test_filter_output_number_correct(self):
-        self.assertEqual(3, len(self.f.get_outputs()))
+        self.assertEqual(3, len(self.f.get_output_names()))
+
+    def test_filter_output_stream_names_equals(self):
+        self.assertEqual(["D", "E", "F"], self.f.get_output_names())
+
+    def test_get_in_streams(self):
+        self.assertEqual(self.s_A, self.f._get_input(0))
+        self.assertEqual(self.s_B, self.f._get_input(1))
+
+    def test_get_out_streams(self):
+        self.assertEqual(self.s_D, self.f._get_output(0))
+        self.assertEqual(self.s_E, self.f._get_output(1))
+        self.assertEqual(self.s_F, self.f._get_output(2))
+
+    def test_get_in_iters(self):
+        self.assertEqual(iter(self.s_A), self.f._get_in_iter(0))
+        self.assertEqual(iter(self.s_B), self.f._get_in_iter(1))
+
+    def test_get_out_iters(self):
+        self.assertEqual(iter(self.s_D), self.f._get_out_iter(0))
+        self.assertEqual(iter(self.s_E), self.f._get_out_iter(1))
+        self.assertEqual(iter(self.s_F), self.f._get_out_iter(2))
+
+    def test_pop_data(self):
+        self.assertEqual(1, self.f._pop_data(0))
+        self.assertEqual(3, self.f._pop_data(1))
+
+    def test_push_data(self):
+        self.f._push_data(5, 0)
+        self.assertEqual(5, self.s_D.__iter__().__next__())
+
+    def test_execute_outputs_closed(self):
+        self.s_D.close()
+        self.s_E.close()
+        self.s_F.close()
+        self.f._on_outputs_closed = MagicMock()
+        self.f.execute()
+        assert self.f._on_outputs_closed.called
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5775588)